### ISSUE TITLE
Not all weakpoints should require dissections.

### ISSUE THREAD

**Issue by Aerin-of-the-Toast** [CleverRaven/Cataclysm-DDA#72450]
### Is your feature request related to a problem? Please describe.

Currently, there's a variety of proficiencies who's sole purpose is making you much more likely to hit all the weakpoints on certain enemies. The main way you acquire these proficiencies is dissecting these enemies. This makes sense _sometimes_, such as dissecting a mutant wasp to learn what's a vital organ and what's just chitin, but it also applies in areas it shouldn't, too. 

Examples being needing to dissect zombies and have anatomical knowledge to know to sweep the legs out to make a humanoid enemy fall over, or attacking the arms to break their grab, or needing to dissect military zombies to learn not to punch them in the ballistic vest even though it is not even a part of them (yet, anyway, dissecting kevlar zombies is a bit different).

You can get these proficiencies through just combat, but it is _exceedingly_ slow.

### Solution you would like.

Not all weakpoints are literally being used as just "a point that is weak," they're also being used as pseudo-techniques. Hence, I suggest splitting weakpoints into 3 types.

1. Anatomical - The current system. Used for weakpoints tied to the enemy's biology. Knowing where the vitals are on anything from moose to a mi-go. I don't think a proficiency should be needed for anything that is basically just a human (i.e. basic zombies), I'm assuming most people have a decent enough idea where their own vitals are, but if one is needed then "Basic Human Anatomy" should be part of a default-enabled background. (Yes, I've met some people completely clueless to their own anatomy, but I'm assuming that to be a minority).
2. Visual - Weakpoints you can _see_. Most prevalent with armor, such as military zombies, but could also include gaps in scar tissue for burned zombies since the burned areas would be different per zombie. This could be tied to perception or eye limb score with varying difficulty per part i.e. it should be really easy to see a helmet on a military zombie that you shouldn't hit but the aforementioned scar tissue would be very difficult to spot gaps in.
3. Technique - Sweeps, grabs, eye gouges. Any "weakpoint" where the difficulty comes less from identifying the weakpoint and more from just pulling it off, and are usually built around doing auxiliary effects (knocking them over, breaking a grab, blinding them) than pure damage. These should likely require a mix of melee skill and perhaps their own set of proficiencies gained solely through combat ("Zombie Fighting" proficiency?).

### Describe alternatives you have considered.

Another idea is removing any "technique"-style weakpoints and just leaving them solely in the martial arts system, but I think that has its own issues. This would simplify it down to just "weakpoints that are or aren't learned through dissection" though.

### Additional context

Who knew this whole time that coroners were the real combat gods? Move over, martial artists, y'all got nothing on Ducky from NCIS.

**Comment by ADekema** [comment]
I agree with it making more sense that some weakpoints are better learned through combet experience than through dissection. I do have a few thoughts on your proposol:

1. There are some enemies that you shouldn't be able to learn the vital weakpoints off from dissection. Stuff like migo, triffid, fungus and some nether creatures have such alien anatomy that a suvivor wouldn't know what they where looking at. This might also go for highly evolved zombies as they seem to function on completely differant constraints and principles than most living organisms.

2. I think it isn't a good idea to replace weakpoints such as arms and legs with only techniques. I think a better solution would be to change techniques from attacks that inflict a specific effect to a attack that guarantees a hit on a certain weakpoint with a greatly increased chance to trigger that weakpoints effect. The whole technique/martial arts systeem is in need of a overhaul anyway.

**Comment by I-am-Erk** [comment]
We just had a long discussion about this with you me and kevin. I am summarizing what I think are the important points.
- Although I theoretically agree with your argument here, I just don't think the difference is enough to bother with the amount of coding and modeling and json and maintenance that would go along with it. Ultimately the differences would be really small, and largely hidden to the player. I think we can abstract it further than that.

However, we all seem to agree that dissection should not be the *main* way to gain weakpoint proficiencies. There are two issues here really. One is that we have fluffed all the weakpoint profs as "biology" knowledge, which I think is silly. The kind of dissection you're doing to learn weakpoints is similar enough to medical dissection as to not matter too much in the game, but the skills you're learning to help combat aren't biology skills. They shouldn't require intro to biology and they shouldn't be named to sound like biology skills. 
- A zombie anatomy proficiency as a descendent of biology is totally appropriate, and should be learned from dissection, and should take much longer to learn, and should not benefit weakpoints.  I also don't think we have any need for this proficiency in game just yet, i just want to make it clear that I don't mind if someone wants to add it when they need it down the road.
- A "zombie combat basics" proficiency might benefit from a dissection or two, should give a benefit to weakpoint hitting, should not require biology as a prereq, and should not be *mainly* learned from dissection.

I think it's important we keep those two things separate.

Aside from restructuring the proficiency dependency tree and the fluff, I see a few ways we could reduce the role of dissection in learning WPs. The main one is just to drop the XP gained from them, and increase the XP gained from fighting, so that dissection feels like a boost rather than the best way to learn. On top of that, I had a few suggestions, any one of which would be fine. I've listed them in increasing order of both complexity and how much I like them.

1) only a certain amount of XP can be gained from dissections in a given proficiency, like 25% or something;
2) dissecting the same monster gives rapidly diminishing returns in proficiency xp, so after your third mon_zombie_scooter you're not learning much;
3) dissection doesn't give proficiency xp, but grants an effect that increases the proficiency gains from that proficiency class for a while - not based on time but based on amount learned, so eg. your next 100 attacks against scooter zombies gain double proficiency xp (basically the same as knowledge/skill but for these proficiencies)

Note that each of these has benefits and drawbacks, and I think any of them would solve the problem just fine. There are probably other options I haven't considered.

**Comment by kevingranade** [comment]
One additional point is that the root problem here isn't necessarily "not everything should require dissections" but rather "we're asking the player to do too many dissections".

For gaining the weak point proficiency that will work on a type of monster in particular, I think it would be reasonable to set it up such that:
1. You can tell by looking at a monster what "body plan" and therefore weak point proficiency it falls under.
2. You can gain that proficiency by dissecting a small number of enemies of that type, possibly just one.
3. optional: you might stop dissecting once you gain this proficiency, so maybe not even a "full" dissection is required?
I think this provides the intended flavor of "actually do some directed research" instead of "just hit it until you figure it out" without burdening the player with a highly repetitive set of actions they get sick of.

**Comment by I-am-Erk** [comment]
Hm yeah, doing it that way would work and be really simple. We'd have "basic zombie knowledge" which is learned fast, either mainly or entirely through dissection depending on the monster type, and offers a very tiny bonus to WPs, and then "zombie combat experience" which has the basic knowledge as a prereq, offers a larger bonus, and does not benefit from dissection at all.

**Comment by zachary-kaelan** [comment]
Even if a human knows their own biology, they may not know what stabbing a particular organ or whatever does in terms of an enemy's combat effectiveness and putting them down. Is it better to stab a zombie in the liver or the kidneys? Does stabbing either do anything at all? Those kinds of questions need to be answered by combat.

I'm not certain it would even be possible to determine how vital an organ was through dissection. It would primarily serve to precisely localize the weak points within the body. How many dissections you need would depend on how experienced you are with dissections, in general (knowing where to start, what things to look for, how to preserve organs and structure, etc; can be trained with books) and for specific body plans or biology types. We can assume the survivor can simply draw up diagrams as they dissect so the benefits should be permanent.

Something dissection-adjacent could be just stabbing corpses in various ways to see what angles allow for hitting internal weak points with minimal resistance from bone and tough cartilage. Also testing with a hammer on the bones to see how to produce the best breaks, or seeing what angles maximize the chance of broken ribs penetrating internal organs. It would be a lot faster than dissection, but would require trial and error on a lot of corpses to find the best weak points. The actual practicality of hitting the weak points on a live target would have to be explored in combat.
